### DISCUSSION TITLE
R513 a

### DISCUSSION THREAD

**Discussion by yasiaee** [oemof/tespy#392] [Q&A] [answered]
I want to use Tespy for modeling heat pump cycles, however the working fluid is R513 a which is not included in tespy. Any help would be much appreciated!

**Comment by fwitte** [comment, marked as answer]
Dear @yasiaee,

TESPy currently relies on CoolProp or REFPROP fluid properties (which can be accessed via the CoolProp wrapper). Since CoolProp does not include R513a you will need to have a REFPROP license (REFPROP does seem to have R513a: https://github.com/usnistgov/REFPROP-issues/issues/452) and create your TESPy network object like so:

```python
Network(fluids=["REFPROP::R513a"], ...)
```

Maybe you can ask the CoolProp developers, if they plan to bring R513a to the table in the future :).

Best

**Reply by artsinli** [reply]
Hey @fwitte
I am trying to do the same thing here but I keep getting
`WARNING:root:Could not find the fluid "R513A" in the fluid property database.`

I have set my REFPROP path variables for both the 32 and 64 bit dlls. Is there anything else I can do to troubleshoot this? Thanks.

**Reply by fwitte** [reply]
Thanks for the report. Unfortunately I do not have a REFPROP license available in this moment, so I cannot make any tests. I will be back from holidays soon and reach out to you. Maybe we need to organize a short online meeting to figure out, what is going on. In the meantime you could try to use the CoolProp functions directly to find if this issue is TESPy oder CoolProp related.

thanks

**Reply by artsinli** [reply]
Sounds good, look forward to talking with you then.

EDIT: It seems like this can do the trick 
```
fluid_list = ["REFPROP::R513A.MIX"]
    my_plant = Network(fluids=fluid_list)
```
However,
`WARNING:root:Could not find values for maximum and minimum pressure.`
Will play around with it to see where I get.

**Reply by yasiaee** [reply]
Hi @artsinli 
I hope you are well! I just want to know that could you connect REFPROP to TESPy? I do not have REFPROP license right now, I want to make sure that I will be able to use it for R513A. Thank you!

**Reply by fwitte** [reply]
@artsinli: Okay, thank you for your feedback. Do you have time on March 20, 17:00 CET? Or if not, what date/time would suit you? I would still like to have a short look at that, really appreciate your help!

@yasiaee: Hi, we will try to test, what the issues are with that fluid and let you know the outcome here. But please be aware: Even if we find that working, I cannot guarantee, you will be able to do what you are trying to do. TESPy is a free and open-source software I develop in my free time. Thank you!
